src/arrayMethodJoin.js

```diff
@@ -6,7 +6,47 @@
 function applyCustomJoin() {
   [].__proto__.join2 = function(separator) {
     // write code here
+    const delimiter = getDelimiter(separator);
+
+    let output = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const current = isValidItem(this[i])
+        ? this[i]
+        : '';
+      const part = !isLastIndex(this, i)
+        ? current + delimiter
+        : current;
+
+      output += part;
+    }
+
+    return output;
   };
 }
 
+function getDelimiter(separator) {
+  if (separator) {
+    return separator.toString();
+  }
+
+  if (separator === '') {
+    return separator;
+  }
+
+  if (separator === null) {
+    return 'null';
+  }
+
+  return ',';
```

Comment:
Instead of this, you can set default value for a separator like this: 

`
[].__proto__.join2 = function(separator = ',' ) {
`